agent/api/task.go

```diff
@@ -210,7 +214,7 @@ func (task *Task) initializeCredentialsEndpoint(credentialsManager credentials.M
 func (task *Task) addNetworkResourceProvisioningDependency() {
 	// TODO check networking mode for the task before doing this
 	for _, container := range task.Containers {
-		if container.Name == emptyHostVolumeName && container.IsInternal {
+		if container.IsInternal {
```

Comment:
Can you add a comment here why we also don't handle `emptyHostVolumeName`, is it because we don't start this container as its `steadState` is `Created`?